### PR TITLE
update empty patterns

### PR DESCRIPTION
**Pull request by SaxonF**
https://github.com/WordPress/gutenberg/assets/1072756/3b8c83b0-5fed-4201-aee4-73ea4ff4843a

## What?
Updates the empty state when no patterns exist.
## Why?
Direct people to create their first pattern.

https://github.com/WordPress/gutenberg/issues/53287

## How?
Updates the no-patterns component and adds a renderTrigger prop to the AddNewPattern component.

## Testing Instructions
1. Go to site editor
2. Open patterns page when no patterns are available
3. Notice empty state in content area on right
4. Click "new pattern" and notice pattern creation modal

### CODE REVIEW

**Review by aaronrobertshaw — DISMISSED** [review]
Nice work @SaxonF 👍 

This PR tests well for me and is a drastic improvement over the simple "no patterns" message.

✅ Creating patterns from dropdown still works
✅ Triggering pattern creation from new empty state works
✅ Empty pattern state displays well when reached via dodgy URL, unmatched filters etc

My one question is should we update this to trigger creating a template part when you are in an empty template part category (e.g. via searching)?

I'm happy to approve this now though as we could refine the template part aspect in a follow-up.

Screenshot 2023-09-20 at 2 40 55 pm

**Comment by talldan** [comment]
Sorry, accidentally closed this! Reopening.

I noticed when testing that the 'All Patterns' category appears selected on the left when visiting the patterns page:
Screenshot 2023-10-02 at 1 43 27 pm

Something about this doesn't feel right to me. If you click on 'All Patterns' at this point you end up with a different view, even though it feels like I had the same thing selected as before.

**Comment by aaronrobertshaw** [comment]
> Something about this doesn't feel right to me. If you click on 'All Patterns' at this point you end up with a different view, even though it feels like I had the same thing selected as before.

After some [constants were refactored](https://github.com/WordPress/gutenberg/pull/54484), a regression was introduced causing the filters not to show. This was fixed in https://github.com/WordPress/gutenberg/pull/54721.

I could replicate that issue on this branch, which was resolved after rebasing this PR.

**Comment by aaronrobertshaw** [comment]
Now https://github.com/WordPress/gutenberg/pull/54906 has merged we'll need to update the e2e tests that currently check for `No patterns found` text for the empty state.

**Comment by talldan** [comment]
@SaxonF This is now looking good to me with that fix. There's one potential convenience feature for users that came to mind, it might be good for the new pattern modal to be pre-filled with the selected category and sync status. What do you think?

**Comment by aaronrobertshaw** [comment]
> There's one potential convenience feature for users that came to mind, it might be good for the new pattern modal to be pre-filled with the selected category and sync status. What do you think?

There's a [draft PR](https://github.com/WordPress/gutenberg/pull/54764) for an alternate duplication flow that reuses the create pattern modal and pre-populates some fields. So it has some overlap with this suggestion. Could it be a separate follow-up?

**Comment by talldan** [comment]
> Could it be a separate follow-up?

Absolutely 👍

File: packages/edit-site/src/components/page-patterns/no-patterns.js [hunk 1]
```diff
@@ -2,11 +2,48 @@
...
-		</div>
+		<VStack
+			spacing={ 6 }
+			alignment="center"
+			className="edit-site-patterns__no-results"
+		>
+			<Icon icon={ symbol } />
+			<VStack spacing={ 3 } alignment="center">
+				<Heading level={ 4 } as="h2">
+					{ __( 'Create a pattern' ) }
```
Comment (kevin940726) [inline packages/edit-site/src/components/page-patterns/no-patterns.js:29]: I wonder if we should still show text suggesting no patterns for accessibility reasons?

**Comment by carolinan** [comment]
I believe this screen is being replaced by an admin view, so is this update still planned to be included in WordPress 6.5?

**Review by t-hamano — COMMENTED** [review]
In #58139, the data view based pattern page has been stabilized. I think you should look for an approach in the data view component for this problem. I submitted #58482.

**Comment by carolinan** [comment]
With 58482 merged, should this be closed?

**Pull request closed without merging**